### PR TITLE
Feature/router optional params

### PR DESCRIPTION
**Pull request by sandeep-vedam**
Added support for optional parameters another approach

### CODE REVIEW

File: src/Router/index.js [hunk 1]
```diff
@@ -237,6 +236,7 @@ const handleHashChange = async override => {
 
   const route = getRouteByHash(hash)
 
+  console.log('ABCDE Route to be navigated =>', route)
```
Comment (ricardo-a-alves-alb) [inline src/Router/index.js:239]: Forgotten?

File: src/Router/index.js [hunk 2]
```diff
@@ -250,17 +250,20 @@ const handleHashChange = async override => {
   request.hash = hash
   request.route = route
 
+  console.log('ABCDE request information =>', request)
```
Comment (ricardo-a-alves-alb) [inline src/Router/index.js:253]: Forgotten?

File: src/Router/index.js [hunk 3]
```diff
@@ -250,17 +250,20 @@ const handleHashChange = async override => {
...
+
   let result = await beforeEachRoute(getActiveHash(), request)
 
   // test if a local hook is configured for the route
   if (result && route.beforeNavigate) {
     result = await route.beforeNavigate(getActiveHash(), request)
   }
-
+  console.log('ABCDE Result beforenavigate =>', result)
```
Comment (ricardo-a-alves-alb) [inline src/Router/index.js:261]: Forgotten?

File: src/Router/index.js [hunk 4]
```diff
@@ -250,17 +250,20 @@ const handleHashChange = async override => {
...
   if (isBoolean(result)) {
     // only if resolve value is explicitly true
     // we continue the current route request
     if (result) {
+      console.log('ABCDE resolve hash change', request)
```
Comment (ricardo-a-alves-alb) [inline src/Router/index.js:266]: Forgotten?

File: src/Router/index.js [hunk 5]
```diff
@@ -299,7 +302,12 @@ const resolveHashChange = request => {
     const component = getComponent(route.path)
     // if a hook is provided for the current route
     if (isFunction(route.hook)) {
-      const urlParams = getValuesFromHash(hash, route.path)
+      let routerPath = route.path
+      if (routerPath.endsWith('?'))
+        routerPath = routerPath.substring(0, routerPath.lastIndexOf('/'))
+      console.log('ABCDE router path before value from hash =>', routerPath, hash)
```
Comment (ricardo-a-alves-alb) [inline src/Router/index.js:308]: Forgotten?

File: src/Router/index.js [hunk 6]
```diff
@@ -299,7 +302,12 @@ const resolveHashChange = request => {
...
+      const urlParams = getValuesFromHash(hash, routerPath)
+      console.log('ABCDE url parameters =>', urlParams)
```
Comment (ricardo-a-alves-alb) [inline src/Router/index.js:310]: Forgotten?

File: src/Router/utils/route.js [hunk 7]
```diff
@@ -120,6 +123,7 @@ export const getRouteByHash = hash => {
         // we only need to the static and regex drive parts
         continue
       } else if (hashPart && routePart.toLowerCase() !== hashPart.toLowerCase()) {
+        // console.log("ABC router is matching =>", routePart, hashPart, false)
```
Comment (ricardo-a-alves-alb) [inline src/Router/utils/route.js:126]: Forgotten?

**Review by ricardo-a-alves-alb — CHANGES_REQUESTED** [review]

**Comment by sandeep-vedam** [comment]
Hey @ricardo-a-alves-alb 
That was a mistake.
Removed all console messages now. :-)

**Comment by sandeep-vedam** [comment]
As the implementation of this PR is done in #362 , closing this PR

**Pull request closed without merging**